### PR TITLE
Features/add medications tables

### PR DESCRIPTION
**Pull request by smakitech**
Temporary files which were used to publish and test database correctly  aren't need to be reviewed:
static_TempPatientInfo.sql
static_TempUsersData.sql
PatientInfo.sql
UsersData.sql
DoctorsView.sql
There is issue with formating, it will be handled.

### CODE REVIEW

**Review by FSuhov — APPROVED** [review]

File: src/Schema/Tables/Medications/PatientMedications.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    -- gives access to PatientIngo table.
+    /*
+    CONSTRAINT FK_PatientMedications_PatientsInfo
+    FOREIGN KEY ( [PatientId] )
+    REFERENCES [dbo].[PatientInfo]( [PatientId] ),
+    */
+
+    CONSTRAINT FK_PatientMedications_Medications
+    FOREIGN KEY ( [MedicationId] )
+    REFERENCES dbo.[Medications]( [MedicationId] )
```
Comment (Anastasiia37) [inline src/Schema/Tables/Medications/PatientMedications.sql]: dbo in []

File: src/Schema/Tables/Medications/Medications.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    [MedicationId] BIGINT NOT NULL IDENTITY(1, 1),
+    [Name]         NVARCHAR(100) NOT NULL,
+    [LatinName]    NVARCHAR(100) NOT NULL,
+    [Type]         NVARCHAR(50) NOT NULL,
+    [Dose]         SMALLINT NOT NULL,
+    [Direction]    NVARCHAR(50) NOT NULL,
+    [Instruction]  NVARCHAR(max) NOT NULL,
+    [IsDeleted]    BIT NOT NULL DEFAULT 0,
+
+    CONSTRAINT PK_Medications
```
Comment (Anastasiia37) [inline src/Schema/Tables/Medications/Medications.sql]: We must stick to one style. We write all constraints in one line

File: src/Schema/Tables/Medications/Medications.sql [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    PRIMARY KEY CLUSTERED ( [MedicationId] ),
+
+    CONSTRAINT UC_Medications
```
Comment (Anastasiia37) [inline src/Schema/Tables/Medications/Medications.sql]: Same

File: src/Schema/Tables/Medications/Medications.sql [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    UNIQUE ( [LatinName], [Type], [Dose] ),
+
+    CONSTRAINT CHK_MedicationsDose
```
Comment (Anastasiia37) [inline src/Schema/Tables/Medications/Medications.sql]: Same

File: src/Schema/Tables/Medications/PatientMedications.sql [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+(
+    [PatientMedicationsId] BIGINT NOT NULL IDENTITY(1, 1),
+    [PatientId]            INT NOT NULL,
+    [UserId]               INT NOT NULL,
+    [MedicationId]         BIGINT NOT NULL,
+    [AssignmentDate]       DATE NOT NULL DEFAULT Getdate(),
+    [Duration]             NVARCHAR(20) NOT NULL,
+    [Notes]                NVARCHAR(max) NULL,
+    -- Сurrent treatment by medicine equals 0, finished equals 1.
+    [Status]               BIT NOT NULL DEFAULT 0
```
Comment (Anastasiia37) [inline src/Schema/Tables/Medications/PatientMedications.sql]: May be not Status, but something like Finished? Than you will not need that comment before column

File: src/Schema/Tables/Medications/PatientMedications.sql [hunk 6]
```diff
@@ -0,0 +1,28 @@
+﻿CREATE TABLE [dbo].[PatientMedications]
+(
+    [PatientMedicationsId] BIGINT NOT NULL IDENTITY(1, 1),
+    [PatientId]            INT NOT NULL,
+    [UserId]               INT NOT NULL,
+    [MedicationId]         BIGINT NOT NULL,
+    [AssignmentDate]       DATE NOT NULL DEFAULT Getdate(),
+    [Duration]             NVARCHAR(20) NOT NULL,
+    [Notes]                NVARCHAR(max) NULL,
```
Comment (Anastasiia37) [inline src/Schema/Tables/Medications/PatientMedications.sql]: MAX

File: src/Schema/Tables/Medications/PatientMedications.sql [hunk 7]
```diff
@@ -0,0 +1,28 @@
+﻿CREATE TABLE [dbo].[PatientMedications]
+(
+    [PatientMedicationsId] BIGINT NOT NULL IDENTITY(1, 1),
+    [PatientId]            INT NOT NULL,
+    [UserId]               INT NOT NULL,
+    [MedicationId]         BIGINT NOT NULL,
+    [AssignmentDate]       DATE NOT NULL DEFAULT Getdate(),
```
Comment (Anastasiia37) [inline src/Schema/Tables/Medications/PatientMedications.sql]: GETDATE()

File: src/Schema/Tables/Medications/PatientMedications.sql [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    [Duration]             NVARCHAR(20) NOT NULL,
+    [Notes]                NVARCHAR(max) NULL,
+    -- Сurrent treatment by medicine equals 0, finished equals 1.
+    [Status]               BIT NOT NULL DEFAULT 0
+
+    CONSTRAINT PK_PatientMedications
```
Comment (Anastasiia37) [inline src/Schema/Tables/Medications/PatientMedications.sql]: Same

File: src/Schema/Tables/Medications/PatientMedications.sql [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    -- This constraint won't available
+    -- without merging which
+    -- gives access to PatientIngo table.
+    /*
+    CONSTRAINT FK_PatientMedications_PatientsInfo
+    FOREIGN KEY ( [PatientId] )
+    REFERENCES [dbo].[PatientInfo]( [PatientId] ),
+    */
+
+    CONSTRAINT FK_PatientMedications_Medications
```
Comment (Anastasiia37) [inline src/Schema/Tables/Medications/PatientMedications.sql]: Same

**Review by Anastasiia37 — APPROVED** [review]

File: src/Schema/Tables/Medications/PatientMedications.sql (hunk 5 above)
Comment (smakitech) [reply src/Schema/Tables/Medications/PatientMedications.sql]: What about IsFinished?

File: src/Schema/Tables/Medications/PatientMedications.sql (hunk 1 above)
Comment (smakitech) [reply src/Schema/Tables/Medications/PatientMedications.sql]: Will edit.

File: src/Schema/Tables/Medications/Medications.sql (hunk 2 above)
Comment (smakitech) [reply src/Schema/Tables/Medications/Medications.sql]: Ok, agree

File: src/Schema/Tables/Medications/Medications.sql (hunk 3 above)
Comment (smakitech) [reply src/Schema/Tables/Medications/Medications.sql]: Agree

File: src/Schema/Tables/Medications/Medications.sql (hunk 4 above)
Comment (smakitech) [reply src/Schema/Tables/Medications/Medications.sql]: Agree

File: src/Schema/Tables/Medications/PatientMedications.sql (hunk 6 above)
Comment (smakitech) [reply src/Schema/Tables/Medications/PatientMedications.sql]: Formatter have eaten. I'll fix it.

File: src/Schema/Tables/Medications/PatientMedications.sql (hunk 7 above)
Comment (smakitech) [reply src/Schema/Tables/Medications/PatientMedications.sql]: Again formatter have eaten.

File: src/Schema/Tables/Medications/PatientMedications.sql (hunk 8 above)
Comment (smakitech) [reply src/Schema/Tables/Medications/PatientMedications.sql]: Agree

File: src/Schema/Tables/Medications/PatientMedications.sql (hunk 9 above)
Comment (smakitech) [reply src/Schema/Tables/Medications/PatientMedications.sql]: Agree

**Pull request merged**